### PR TITLE
Feature/preserve data

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- [x] Make sure that you used the correct Gitflow.
- [x] Make sure that you document your work in a professional way.
### Project requirements
- [x] On exit from your app save all the data (books, people, and rentals) into JSON files.
- [x]  One per type of data you are storing and named accordingly (e.g. rentals.json)
- [x] On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -11,4 +12,26 @@ def add_rental(rental)
...
+
+  def self.save_books_to_json(books)
+    books_data = books.map(&:to_hash)
+    File.write('books.json', JSON.pretty_generate(books_data))
+  end
+
+  def self.load_books_from_json
+    books_data = JSON.parse(File.read('books.json'))
+    books_data.map { |data| Book.new(data['title'], data['author']) }
+  end
```
Comment (codepantha) [inline book.rb]: - [ ] Please handle the case when the `book.json` file is missing. Currently, if I delete the `books.json` file and try to start the app, an error occurs as seen below 👇. The app should work and show no errors.
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Benawi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Data preserved on exit, and loaded on startup 📌 
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
### Changes ♻️
- [x] Handle the case when the book.json file is missing.

**Review by OyePriscilla — APPROVED** [review]
Hi @Benawi, @ProgramKingAbel and @sahlminkok 👏 👏,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
App start up as expected when one or more of the files are missing ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**